pages/tutorials/3.0/window/events.md

```diff
@@ -7,259 +7,329 @@
+The data associated with these events contains the code of the pressed/released button, as well as the current position of the mouse cursor.
 
 ```cpp
-if (event.type == sf::Event::MouseButtonPressed)
+if (const auto* mouseButtonPressed = event->getIf<sf::Event::MouseButtonPressed>())
 {
-    if (event.mouseButton.button == sf::Mouse::Right)
+    if (mouseButtonPressed->button == sf::Mouse::Button::Right)
     {
         std::cout << "the right button was pressed" << std::endl;
-        std::cout << "mouse x: " << event.mouseButton.x << std::endl;
-        std::cout << "mouse y: " << event.mouseButton.y << std::endl;
+        std::cout << "mouse x: " << mouseButtonPressed->position.x << std::endl;
+        std::cout << "mouse y: " << mouseButtonPressed->position.y << std::endl;
     }
 }
 ```
 
 ## The MouseMoved event
 
 The `sf::Event::MouseMoved` event is triggered when the mouse moves within the window.
+This event is triggered even if the window isn't focused.
+However, it is triggered only when the mouse moves within the inner area of the window, not when it moves over the title bar or borders.
+
+The data associated with this event contains the current position of the mouse cursor relative to the window.
+
+```cpp
+if (const auto* mouseMoved = event->getIf<sf::Event::MouseMoved>())
+{
+    std::cout << "new mouse x: " << mouseMoved->position.x << std::endl;
+    std::cout << "new mouse y: " << mouseMoved->position.y << std::endl;
+}
+```
+
+## The MouseMovedRaw event
+
+The `sf::Event::MouseMovedRaw` event is triggered when the mouse moved within the window even if the mouse is moved a distance too small to perceive.
 
-This event is triggered even if the window isn't focused. However, it is triggered only when the mouse moves within the inner area of the window, not when it moves over the title bar or borders.
+While the MouseMoved position value is dependent on the screen resolution, raw data is not.
+If the physical mouse is moved too little to cause the screen cursor to move at least a single pixel, no MouseMoved event will be generated.
+In contrast, any movement information generated by the mouse independent of its sensor resolution will always generate a `MouseMovedRaw` event.
 
-The member associated with this event is `event.mouseMove`, it contains the current position of the mouse cursor relative to the window.
+The data associated with this event contains the change in position of the mouse cursor relative to the window.
 
 ```cpp
-if (event.type == sf::Event::MouseMoved)
+if (const auto* mouseMovedRaw = event->getIf<sf::Event::MouseMovedRaw>())
 {
-    std::cout << "new mouse x: " << event.mouseMove.x << std::endl;
-    std::cout << "new mouse y: " << event.mouseMove.y << std::endl;
+    std::cout << "new mouse x: " << mouseMoved->delta.x << std::endl;
+    std::cout << "new mouse y: " << mouseMoved->delta.y << std::endl;
 }
 ```
 
+Currently, raw mouse input events will only be generated on Windows and Linux.
```

Comment:
I guess this could either be a `!!! warning` or `!!! note`. Maybe we need to come up with a clearer definition when to use what.